app-web/gatsby/onCreateNode.js

```diff
@@ -38,5 +40,48 @@ module.exports = ({ node, actions }) => {
       name: 'location',
       value: node.venue ? node.venue.address_1 : '',
     });
+  } else if (node.internal.type === 'MarkdownRemark') {
+    let title = node.frontmatter.title;
+    const ast = remark.parse(node.internal.content);
+    //if our title is blank, visit will search through the content for a usable and reasonable title
+    visit(ast, 'heading', node => {
+      // is title blank and is node on first line and a h1 or h2?
+      if (title === '' && (node.depth === 1 || node.depth === 2)) {
+        // accept headers up to 3rd line of markdown file
+        if (node.position.start.line < 3) {
+          title = node.children[0].value;
+        }
+      }
+    });
+
+    createNodeField({
+      node,
+      name: 'title',
+      value: node.frontmatter.title ? node.frontmatter.title : title,
+    });
+    createNodeField({
+      node,
+      name: 'description',
+      value: node.frontmatter.description ? node.frontmatter.description : '',
+    });
+    createNodeField({
+      node,
+      name: 'image',
+      value: node.frontmatter.image ? node.frontmatter.image : '',
+    });
+    createNodeField({
+      node,
+      name: 'labels',
+      value: !node.frontmatter.labels
```

Comment:
lets abstract this code away into a simple var, multiple ternarys can be hard to read